### PR TITLE
[DOCS][8.3] Create CSPM / KSPM page

### LINKED ISSUE #2044: [DOCS] Enable feature flag for KSPM
We need to add a [technical preview](https://www.elastic.co/guide/en/security/current/sec-experimental-intro.html) doc that shows how to enable the feature flag for KSPM (Kubernetes Security Posture Management). 

Google doc with instructions is [here](https://docs.google.com/document/d/10sMfGr5zPPuRXYCJJzwJe3h2FcwOy6wk3w1cW5Lhek0/edit?usp=sharing). 

Add a note at the top that says: 
> "This feature is available for Elastic Stack versions 8.3.0 and newer." 

@tinnytintin10 will provide us with the appropriate intro paragraph.

### PR DESCRIPTION
**Pull request by benironside**
Resolves #2044 by creating a page with instructions for enabling CSPM / KSPM.

[Preview](https://security-docs_2089.docs-preview.app.elstc.co/guide/en/security/master/security-posture-management.html)

### CODE REVIEW

**Comment by tinnytintin10** [comment]
@benironside I just updated some things, please take a look and lmk what you think. Feel free to reword/restructure some of the things I added. 

I will review again when you're ready to merge to master 🙌🏾

**Comment by benironside** [comment]
[elasticmachine run elasticsearch-ci/docs rebuild

**Comment by benironside** [comment]
@elasticmachine run elasticsearch-ci/docs rebuild

File: docs/experimental-features/security-posture-management.asciidoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+First, enable the KSPM flag for your deployment:
+
+* From the left-side navigation menu in Kibana, click on **Manage this deployment**.
+
+image::images/kspm-1.png[The Manage deployment button]
+
+* Under "My deployment", select **Kibana**.
+* Under "Instances", click the three-dot menu next to your instance, then click **Edit configuration**.
+* Click **Edit user settings** in the upper right of the page.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:25]: Is this for Kibana?  You can click on similarly hyperlinked text for Elasticsearch and Enterprise Search.
Screen Shot 2022-06-24 at 3 02 54 PM


Screen Shot 2022-06-24 at 3 03 03 PM

File: docs/experimental-features/security-posture-management.asciidoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* Add the following line to the User settings: `xpack.cloudSecurityPosture.enabled: true`.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:26]:
```suggestion
* Add `xpack.cloudSecurityPosture.enabled: true` to the User settings.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+* Next, click the **Back** button under User settings, and then click **Save** in the lower left corner, and confirm your changes.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:27]:
```suggestion
* Click **Back** then save your Kibana user settings.
```

Comment (nastasha-solomon) [reply docs/experimental-features/security-posture-management.asciidoc:27]: Here are the Kibana instructions if you want to model these similarly: https://www.elastic.co/guide/en/cloud/current/ec-manage-kibana-settings.html#ec-manage-kibana-settings

File: docs/experimental-features/security-posture-management.asciidoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+NOTE: It may take up to two minutes for the changes to take effect.
+
+Navigate back to Kibana. On the left-side navigation menu, open the Security submenu. Click on **Cloud Security** at the bottom of the menu.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:31]:
```suggestion
Navigate back to Kibana. Open the main menu, then go to the Security section and click **Cloud Security**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+image::images/kspm-2.png[The Cloud Security button on the main menu, width=160]
+
+Follow the on-screen prompts to set up data ingestion.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:35]:
```suggestion
Follow the prompts to set up data ingestion.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+[[enable-kspm-on-selfhosted]]
+[discrete]
+==== Enable CSPM and KSPM for self-hosted deployments
+Edit the kibana.yaml file found in `/config/kibana.yaml/` to include ` `xpack.cloudSecurityPosture.enabled: true`
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:40]:
```suggestion
Edit the `kibana.yaml` file found in `/config/kibana.yaml/` to include ` `xpack.cloudSecurityPosture.enabled: true`
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+Navigate back to Kibana. On the left-side navigation menu, open the Security submenu. Click on **Cloud Security** at the bottom of the menu.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:42]:
```suggestion
Navigate back to Kibana. Open the main menu, then go to the Security section and click **Cloud Security**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+image::images/kspm-2.png[The Cloud Security button on the main menu, width=160]
+
+Follow the on-screen prompts to set up data ingestion.
```
Comment (nastasha-solomon) [inline docs/experimental-features/security-posture-management.asciidoc:46]:
```suggestion
Follow the prompts to set up data ingestion.
```

**Review by nastasha-solomon — COMMENTED** [review]
Left some suggestions for slight adjustments. Thanks for writing this up, looks good!

File: docs/experimental-features/security-posture-management.asciidoc (hunk 1 above)
Comment (benironside) [reply docs/experimental-features/security-posture-management.asciidoc:25]: Yes I think it's got to be the Kibana user settings. Does the way it's written seem like it could mean to click "Edit user settings" next to Enterprise search or Elasticsearch instead?

Comment (nastasha-solomon) [reply docs/experimental-features/security-posture-management.asciidoc]: Yep, I would adjust it like so:

. In the Kibana section, click **Edit user settings**.

File: docs/experimental-features/security-posture-management.asciidoc [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+[discrete]
+=== Enable security posture management
+
+The steps to enable this feature differ between Elastic cloud and self-hosted deployments.
+
+[[enable-kspm-on-cloud]]
+[discrete]
+==== Enable KSPM for cloud deployments
+
+First, enable the KSPM flag for your deployment:
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:17]: Echoing @nastasha-solomon's comment about making _all_ of these steps numbered, and sub-numbered/alpha. It's hard to follow with actions in different formats, when they're all just steps to follow. Something like:

1. Enable the KSPM flag:
   a. From Kibana main menu, click Manage this...
   b. Under My deployment...
   [ . . . ]
   f. Click Back then save...

2. Navigate back to Kibana...

3. Follow the prompts...

File: docs/experimental-features/security-posture-management.asciidoc [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+Navigate back to Kibana. Open the main menu, then go to the Security section and click **Cloud Security**.
+
+image::images/kspm-2.png[The Cloud Security button on the main menu, width=160]
+
+Follow the prompts to set up data ingestion.
+
+[[enable-kspm-on-selfhosted]]
+[discrete]
+==== Enable KSPM for self-hosted deployments
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:39]: Similar thing here with numbered steps:
1. Edit the kibana.yml...
2. Navigate back...
3. Follow the prompts...

File: docs/experimental-features/security-posture-management.asciidoc [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+The steps to enable this feature differ between Elastic cloud and self-hosted deployments.
+
+[[enable-kspm-on-cloud]]
+[discrete]
+==== Enable KSPM for cloud deployments
+
+First, enable the KSPM flag for your deployment:
+
+* From {kib}, open the main menu and click on **Manage this deployment**.
```
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:19]:
```suggestion
* From {kib}, open the main menu and click  **Manage this deployment**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+image::images/kspm-1.png[The Manage deployment button]
+
+1. Under `My deployment`, select **Kibana**.
```
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:23]:
```suggestion
1. Under **My deployment**, select **Kibana**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+Navigate back to Kibana. Open the main menu, then go to the Security section and click **Cloud Security**.
+
+image::images/kspm-2.png[The Cloud Security button on the main menu, width=160]
+
+Follow the prompts to set up data ingestion.
+
+[[enable-kspm-on-selfhosted]]
+[discrete]
+==== Enable KSPM for self-hosted deployments
+Edit the `kibana.yml` file typically found in `/config/kibana.yml/` to include `xpack.cloudSecurityPosture.enabled: true`
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:40]: A few punctuation suggestions. I think ending a path with `/` could suggest that the last segment is the name of a folder, but I assume it's a file here. 
```suggestion
Edit the `kibana.yml` file (typically found at `/config/kibana.yml`) to include `xpack.cloudSecurityPosture.enabled: true`.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+Navigate back to Kibana. Open the main menu, then go to the Security section and click **Cloud Security**.
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:42]: Same as above:
```suggestion
Navigate back to {kib}. From the main menu, go **Security** -> **Cloud Security**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+==== Enable KSPM for cloud deployments
+
+First, enable the KSPM flag for your deployment:
+
+* From {kib}, open the main menu and click on **Manage this deployment**.
+
+image::images/kspm-1.png[The Manage deployment button]
+
+1. Under `My deployment`, select **Kibana**.
+2. Under "Instances", click the three-dot menu next to your instance, then click **Edit configuration**.
```
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:24]: @joepeeples correct me if it should be quotes instead of bold -- can't remember what we've done traditionally. 
```suggestion
2. Under **Instances**, click the three-dot menu next to your instance, then click **Edit configuration**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+3. Click **Edit user settings** in the upper right of the page.
+4. Add `xpack.cloudSecurityPosture.enabled: true` to the User settings.
+5. Click **Back** then save your Kibana user settings.
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:27]:
```suggestion
3. Click **Edit user settings** in the upper right.
4. Add `xpack.cloudSecurityPosture.enabled: true` to the User settings.
5. Click **Back**, then save your {kib} user settings.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+First, enable the KSPM flag for your deployment:
+
+* From {kib}, open the main menu and click on **Manage this deployment**.
+
+image::images/kspm-1.png[The Manage deployment button]
+
+1. Under `My deployment`, select **Kibana**.
+2. Under "Instances", click the three-dot menu next to your instance, then click **Edit configuration**.
+3. Click **Edit user settings** in the upper right of the page.
```
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:25]:
```suggestion
3. Click **Edit user settings** in the upper-right of the page.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 18]
```diff
@@ -0,0 +1,46 @@
+[[security-posture-management]]
+== Cloud security posture management
+
+NOTE: This feature requires Elastic Stack version 8.3.0 or later.
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:4]:
```suggestion
NOTE: This feature requires {stack} version 8.3.0 or later.
```

File: docs/experimental-features/security-posture-management.asciidoc (hunk 12 above)
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:23]:
```suggestion
1. Under `My deployment`, select **{kib}**.
```

File: docs/experimental-features/security-posture-management.asciidoc (hunk 11 above)
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:19]: This is technically step 1, right? Shouldn't this be a number instead of a bullet? 
```suggestion
* From {kib}, open the main menu and click on **Manage this deployment**.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Under `My deployment`, select **Kibana**.
+2. Under "Instances", click the three-dot menu next to your instance, then click **Edit configuration**.
+3. Click **Edit user settings** in the upper right of the page.
+4. Add `xpack.cloudSecurityPosture.enabled: true` to the User settings.
+5. Click **Back** then save your Kibana user settings.
+
+NOTE: It may take up to two minutes for the changes to take effect.
+
+Navigate back to Kibana. Open the main menu, then go to the Security section and click **Cloud Security**.
```
Comment (joepeeples) [inline docs/experimental-features/security-posture-management.asciidoc:31]: Fairly common way to shorten something like this:
```suggestion
Navigate back to {kib}. From the main menu, go **Security** -> **Cloud Security**.
```

File: docs/experimental-features/security-posture-management.asciidoc (hunk 13 above)
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:40]:
```suggestion
Edit the `kibana.yml` file, typically found in `/config/kibana.yml/` to include `xpack.cloudSecurityPosture.enabled: true`.
```

**Review by joepeeples — COMMENTED** [review]
Great, clean approach to a big new feature! I added a mix of suggestions on line-edits and macro formatting, thanks!

File: docs/experimental-features/security-posture-management.asciidoc (hunk 14 above)
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:42]: `Go back` is another alternative. 
```suggestion
Return to {kib}. From the main menu, go to **Security** -> **Cloud Security**.
```

File: docs/experimental-features/security-posture-management.asciidoc (hunk 19 above)
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:31]:
```suggestion
Return to {kib}. From the main menu, go to **Security** -> **Cloud Security**.
```

File: docs/experimental-features/security-posture-management.asciidoc (hunk 16 above)
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:27]:
```suggestion
5. Click **Back**, then save your {kib} user settings.
```

File: docs/experimental-features/security-posture-management.asciidoc [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+First, enable the KSPM flag for your deployment:
+
+* From {kib}, open the main menu and click on **Manage this deployment**.
+
+image::images/kspm-1.png[The Manage deployment button]
+
+1. Under `My deployment`, select **Kibana**.
+2. Under "Instances", click the three-dot menu next to your instance, then click **Edit configuration**.
+3. Click **Edit user settings** in the upper right of the page.
+4. Add `xpack.cloudSecurityPosture.enabled: true` to the User settings.
```
Comment (jmikell821) [inline docs/experimental-features/security-posture-management.asciidoc:26]:
```suggestion
4. Add `xpack.cloudSecurityPosture.enabled: true` to the user settings.
```

**Review by jmikell821 — COMMENTED** [review]
@benironside left some comments for your review. Your images appear to be "floating" - I would wrap a 1-2px dark border around each of them to give them some definition.

File: docs/experimental-features/experimental-features-intro.asciidoc [hunk 21]
```diff
@@ -1,8 +1,9 @@
 [[sec-experimental-intro]]
 = Technical preview
 
-The following features in this section are experimental and may be changed or removed completely in a future release. Elastic will make a best effort to fix any issues, but experimental features are not subject to the support of official GA features.
+The features in this section are experimental and may be changed or removed completely in future releases. Elastic will make a best effort to fix any issues, but experimental features are not supported to the same level as official generally available (GA) features.
```
Comment (benironside) [inline docs/experimental-features/experimental-features-intro.asciidoc:4]:
```suggestion
The features in this section are experimental and may be changed or removed completely in future releases. Elastic will make a best effort to fix any issues, but experimental features are not supported to the same level as generally available (GA) features.
```

File: docs/experimental-features/security-posture-management.asciidoc (hunk 15 above)
Comment (benironside) [reply docs/experimental-features/security-posture-management.asciidoc:21]: going with bold for now

**Review by tinnytintin10 — APPROVED** [review]
LGTM
